test/jdk/java/nio/channels/DatagramChannel/SelectWhenRefused.java

```diff
@@ -49,43 +55,62 @@ public static void main(String[] args) throws IOException {
             dc.configureBlocking(false);
             dc.register(sel, SelectionKey.OP_READ);
 
-            /* Test 1: not connected so ICMP port unreachable should not be received */
-            sendDatagram(dc, refuser);
-            int n = sel.select(2000);
-            if (n > 0) {
-                sel.selectedKeys().clear();
-                // BindException will be thrown if another service is using
-                // our expected refuser port, cannot run just exit.
-                DatagramChannel.open().bind(refuser).close();
-                throw new RuntimeException("Unexpected wakeup");
-            }
+            for (;;) {
```

Comment:
treat each test scenario individually wrt retry and set an upper bound on the number of potential retries ?

Reply:
Having an upper bound - e.g. 3 and failing if we reach the upper bound would be prudent yes :-)